aman/src/pages/Login.jsx

```diff
@@ -0,0 +1,152 @@
+import { useState } from "react";
+import { z } from "zod";
+import { useNavigate } from "react-router-dom";
+import axios from "axios";
+import withLoading from "../components/withLoading";
+import { LuEyeOff } from "react-icons/lu";
+import { LuEye } from "react-icons/lu";
+import Input from "../components/Input";
+import apiClient from "../api/ApiClient";
+
+const LoadingButton = withLoading();
+
+const schema = z.object({
+  email: z.string().email("Must be a valid email address"),
+  password: z.string().min(6, "Password must be 6 or more characters long"),
+});
+
+const Login = ({ handleClick, isLoading, setIsLoading }) => {
+  const [showPsswrd, setshowPsswrd] = useState(false);
+  const [formData, setFormData] = useState({ email: "", password: "" });
+  const [error, setError] = useState({});
+  const [wrongError, setWrongError] = useState("");
+  const navigate = useNavigate();
+
+  const handleChange = (e) => {
+    setFormData({ ...formData, [e.target.name]: e.target.value });
+  };
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    const result = schema.safeParse({
+      ...formData,
+    });
+    if (!result.success) {
+      const formattedErrors = result.error.format();
+      setError(formattedErrors);
+    } else {
+      setError({});
+      setIsLoading(true);
+      try {
+        await handleLogin();
+      } finally {
+        setIsLoading(false);
+      }
+    }
+  };
+
+  const handleShow = () => {
+    setshowPsswrd(!showPsswrd);
+  };
+
+  const handleLogin = async () => {
+    try {
+      const response = await apiClient.post("/auth/login", formData);
+      const data = await response.data;
+      console.log(data);
```

Comment:
You can remove console logs wherever not required.

Reply:
Done